javascript/ql/src/meta/analysis-quality/TaintSteps.ql

```diff
@@ -0,0 +1,24 @@
+/**
+ * @name Taint steps
+ * @description The number of default taint steps.
+ * @kind metric
+ * @metricType project
+ * @metricAggregate sum
+ * @tags meta
+ * @id js/meta/taint-steps
+ */
+
+import javascript
+import CallGraphQuality
+
+class BasicTaintConfiguration extends TaintTracking::Configuration {
+  BasicTaintConfiguration() { this = "BasicTaintConfiguration" }
+}
+
+predicate relevantStep(DataFlow::Node pred, DataFlow::Node succ) {
+  any(BasicTaintConfiguration cfg).isAdditionalFlowStep(pred, succ) and
+  not pred.getFile() instanceof IgnoredFile and
```

Comment:
I think we should count it if we step *into* a file that we do not ignore.
Ideally, we can track taint flow through a minimised library into the ordinary client code.

Reply:
Hm, one of the issues I wanted to avoid is having to triage results involving minified files. That's also something that can be solved at the UI level, like the dist-compare reports currently do for ordinary alerts, and if we do that maybe we won't need `IgnoredFile` at all. Even as I write this it seems obvious that that's what we should do, but maybe we can keep it like this until that's ready?

Reply:
OK.